lib/APPTurtle/src/Transports/CustomRosTransportTcp.cpp

```diff
@@ -0,0 +1,219 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Custom Micro-ROS transport using TCP over Arduino WifiClient.
+ * @author Norbert Schulz <github@schulznorbert.de>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "CustomRosTransportTcp.h"
+
+#include <Logging.h>
+#include <SimpleTimer.hpp>
+#include <Board.h>
+
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+/**
+ * Name of implemented protocol.
+ */
+static const String gProtocolName("TCP");
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+/******************************************************************************
+ * Protected Methods
+ *****************************************************************************/
+
+/******************************************************************************
+ * Private Methods
+ *****************************************************************************/
+
+bool CustomRosTransportTcp::open(void)
+{
+    bool      isOpen = false;
+
+    if (!m_tcpClient.connect(m_address, m_port))
+    {
+        LOG_ERROR("TCP connect error");
+    }
+    else
+    {
+        isOpen = true;
+    }
+
+    return isOpen;
+}
+
+bool CustomRosTransportTcp::close()
+{
+    m_tcpClient.stop();
+    LOG_DEBUG("tcp stream closed.\n");
+    return true;
+}
+
+size_t CustomRosTransportTcp::write(const uint8_t* buffer, size_t size, uint8_t* errorCode)
+{
+    size_t sent = 0U;
+
+    if ((nullptr == buffer) || (0U == size) || (nullptr == errorCode))
+    {
+        LOG_ERROR("One or more parameters are invalid.");
+    }
+    else
+    {
+        uint8_t lengthPrefix[0];
```

Comment:
Array of 0 causes in the following lines out of bounds writes.

Reply:
Ups, did ths wrong during pre-checkin cleanup. Odd that the compiler didn't care ...